h2/src/main/org/h2/value/Transfer.java

```diff
@@ -540,13 +540,7 @@ public void writeValue(Value v) throws IOException {
             ValueArray va = (ValueArray) v;
             Value[] list = va.getList();
             int len = list.length;
-            Class<?> componentType = va.getComponentType();
-            if (componentType == Object.class) {
-                writeInt(len);
-            } else {
-                writeInt(-(len + 1));
-                writeString(componentType.getName());
-            }
+            writeInt(len);
```

Comment:
this is a protocol change and will break old clients talking to newer servers and vice versa

Reply:
This place is fine, but read method must skip the class name, if any.

Reply:
Good catch, BTW. A fix of `readValue()` is in #2189.